File: Euler2.java

  /**
   * This program prints the sum of all even number
   * in fibonacci series below 4 million (4000000)
   */

   public class Euler2 {

   	  public static void main(String[] args) {

   	  	  int fibA = 0;
   	  	  int fibB = 1;
   	  	  int fibC = 0;
          int total = 0;

          while(true) {

          	  fibC = fibA + fibB; // sum previous two terms to get next term
          	  fibA = fibB; // advance to next term
          	  fibB = fibC;

          	  if(fibC > 4000000) 
          	  	break;

          	  if(fibC % 2 == 0) 
          	  	total += fibC;

          }

          System.out.println(" Sum of all even numbers not exceeding 4 million in fibonacci series is " + total);
   	  }
   }

File: euler_2.py
# This program prints sum of even fibonnaci series number below 4 million
a = 0
b = 1
s = 0

while b < 400001:
    if (a + b) % 2 == 0:
       s += (a + b)
    b = (a + b)   
    a = (b - a)

print 'THE SUM OF ALL EVEN FIBONACCI NUMBERS BELOW 4 MILLION IS ',
print s

